app/assets/javascripts/external-scripts.js

```diff
@@ -30,43 +42,28 @@ ExternalScripts.init = function(_label, ctx) {
       this._script_labels.push(_label);
     }
     catch(e) {
-      this.externalScriptError(e, ctx);
+      this._externalScriptError(e, ctx);
     }
   }
   else {
-    this.notRegistered(_label);
+    this._notRegistered(_label);
   }
 }
 
-ExternalScripts.config = function(_label, ctx) {
-  var constructor = this._constructors[_label];
-  var script = null;
-  if (typeof constructor == 'function') {
-    if (typeof constructor.config == 'function'){
-      try {
-        script = constructor.config(ctx);
-      }
-      catch(e) {
-        this.externalScriptError(e, ctx);
-      }
-    }
-    else {
-      console.log("No configuration for " + _label);
-    }
-  }
-  else {
-    this.notRegistered(_label);
-  }
-}
-
-
-ExternalScripts.register = function(_label, _constructor) {
-  if(typeof _constructor == 'function') {
-    if(this._constructors[_label]) {
+/**************************************************************
+ @function register
+ Register a new external script as `label` with `_class `
+ @arg label - the identifier of the sctipt
```

Comment:
The label is used as a key. When a script registers itself it uses this value. LARA uses this value both to figure out which scripts to put on the page, but also what constructors to call. We could probably devise a different mechanism to coordinate this. We could for example use javascript ClassNames -- and just let the script manage it. But since there is the possibility of name-space collisions, we probably want to the admin to manage it in some way?

In short, I agree this seems redundant on the surface, but I believe some mechanism like this would be required.

Reply:
Right, looking at the current implementation, I can see what you mean. However, I think it might not be necessary in general.

An author selects a script using pulldown menu. It uses a "name" (which is different from "label"). So, LARA knows which script to add to the page (by adding <script> tag).

Then, plugin calls `ExternalScript.register(MyPluginClass)` providing a constructor to LARA. LARA saves this constructor. At this point, LARA could immediately initialize the plugin too and it should solve the problem. 
Or we do it later - LARA simply iterates over all saved constructors and initializes all the plugins. But we'd need to make sure that it happens after they are all added to the page (perhaps including them in <head> and calling that at the end of <body>, or something like that). Not saying it's super important, but I feel it's possible to simplify that a bit.